homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,176 @@
+    STATE_PRESET_SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
+}
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    _attr_hvac_modes = [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODES[
+            cast(
+                str, self.executor.select_state(OverkizState.CORE_DEROGATION_ACTIVATION)
+            )
+        ]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        if not self.current_temperature or not self.target_temperature:
+            return CURRENT_HVAC_IDLE
+        if self.current_temperature < self.target_temperature:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return OVERKIZ_TO_PRESET_MODES[
+                cast(
+                    str,
+                    self.executor.select_state(
+                        OverkizState.SOMFY_THERMOSTAT_HEATING_MODE
+                    ),
+                )
+            ]
+        return OVERKIZ_TO_PRESET_MODES[
+            cast(
+                str,
+                self.executor.select_state(
+                    OverkizState.SOMFY_THERMOSTAT_DEROGATION_HEATING_MODE
+                ),
+            )
+        ]
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if temperature := self.temperature_device.states[OverkizState.CORE_TEMPERATURE]:
+            return cast(float, temperature.value)
+        return None
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            if self.preset_mode == PRESET_NONE:
+                return None
```

Comment:
Why is there no target temperature when the preset is not set? Should a preset not be optional.

Reply:
Good question. @vlebourl Do you rembember why you did this? IMO we can remove this condition. According to https://github.com/home-assistant/core/pull/67169/files#diff-7b52214972019e7783fcb8b4528bc473433cd1f1d6476d57c2203f6d3844581bR99 when `self.hvac_mode == HVAC_MODE_AUTO ` the preset mode is set to `select_state(OverkizState.SOMFY_THERMOSTAT_HEATING_MODE)`

Reply:
It was a long time ago, but I believe it's just a "return back to what was set before a preset was used". The target temperatures for presets and "no preset" modes are stored in different attributes of the thermostat, so setting preset to NONE automatically gets back to the previously targeted temperature.